src/property.ts

```diff
@@ -984,15 +917,17 @@ export function Property$init(property: Property, obj: Entity, val: any): void {
 }
 
 function Property$ensureInited(property: Property, obj: Entity): void {
-	// Determine if the property has been initialized with a value
-	// and initialize the property if necessary
+	// Determine if the property has been initialized with a value and initialize the property if necessary
 	if (!obj.__fields__.hasOwnProperty(property.name)) {
-		// Mark the property as pending initialization
 		Property$pendingInit(obj, property, true);
 
 		// Do not initialize calculated properties. Calculated properties should be initialized using a property get rule.
 		if (!property.isCalculated) {
 			Property$init(property, obj, Property$getInitialValue(property));
+
+			// Mark the property as pending initialization if it still has no underlying value to allow default calculation rules to run for it
+			if (obj.__fields__[property.name] === null)
+				Property$pendingInit(obj, property, true);
```

Comment:
I adapted this from exoweb:
![image](https://user-images.githubusercontent.com/4603206/120719411-377dda80-c498-11eb-9923-521877606a26.png)

Reply:
Oh, now I see that you changed the `isCalculated` logic to not set it when its a default value calculation 😁